src/coreComponents/mpiCommunications/NeighborCommunicator.cpp

```diff
@@ -289,9 +289,10 @@ inline int GhostSize( NodeManager & nodeManager, arrayView1d< localIndex const >
   bufferSize += edgeManager.packUpDownMapsSize( edgeAdjacencyList );
   bufferSize += faceManager.packUpDownMapsSize( faceAdjacencyList );
   bufferSize += elemManager.PackUpDownMapsSize( elementAdjacencyList );
-  bufferSize += nodeManager.packSize( {}, nodeAdjacencyList, 0 );
-  bufferSize += edgeManager.packSize( {}, edgeAdjacencyList, 0 );
-  bufferSize += faceManager.packSize( {}, faceAdjacencyList, 0 );
+  parallelDeviceEvents noEvents;
```

Comment:
error check empty

Reply:
I see a bunch of these comments, error check that `noEvents` contains no events? That seems like something that should be in a unit tests and not here. Like each time you allocate an `array1d` you don't check that it's `size() == 0`.

Reply:
Ahhh I see, and so Randy wanted to check check that they were empty after passing to the functions.

Reply:
Wait...did I miss something? We expect `noEvents` to be empty, and if it isn't empty, then the events will be async'ed. However we were expecting nothing out of `noEvents`. I don't know that this is actually a problem, but it seems like we should enforce the intent.

What is the difference between "check" if their empty from "check check"?

Reply:
Don't think you missed anything, no. It's just a preference between a (currently) erroneous call to `waitAllDeviceEvents` and an error check against the size of noEvents. Either will guard against a future error.